From 92f419af578bf8162022482ab40b9fca56cad8d3 Mon Sep 17 00:00:00 2001
From: SupervisedThinking <supervisedthinking@gmail.com>
Date: Mon, 17 Oct 2022 17:04:32 +0200
Subject: [PATCH] Audiomanager: adjust music folder paths for LibreELEC-RR

---
 es-core/src/AudioManager.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/es-core/src/AudioManager.cpp b/es-core/src/AudioManager.cpp
index 90b8da17ef..d3690630e4 100644
--- a/es-core/src/AudioManager.cpp
+++ b/es-core/src/AudioManager.cpp
@@ -150,13 +150,13 @@ void AudioManager::playRandomMusic(bool continueIfPlaying)
 
 	std::vector<std::string> musics;
 	
-	// check in RetroPie music directory
+	// check in storage music directory
 	if(musics.empty())
-		getMusicIn(Utils::FileSystem::getHomePath() + "/RetroPie/roms/music", musics);
+		getMusicIn("/storage/music/emulationstation", musics);
   
 	// check in system sound directory
 	if(musics.empty())
-		getMusicIn("/opt/retropie/music", musics);
+		getMusicIn("/usr/share/emulationstation/music", musics);
   
 	// check in .emulationstation/music directory
 	if(musics.empty())
