From 46cca9e73cff21284e9d35937b6cd312599f74d1 Mon Sep 17 00:00:00 2001
From: SupervisedThinking <supervisedthinking@gmail.com>
Date: Tue, 13 Dec 2022 00:35:31 +0100
Subject: [PATCH] platform: add .wua to wiiu files

---
 src/platform.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/platform.cpp b/src/platform.cpp
index 70954ff4..437f8869 100644
--- a/src/platform.cpp
+++ b/src/platform.cpp
@@ -526,7 +526,7 @@ QString Platform::getFormats(QString platform, QString extensions, QString addEx
   } else if(platform == "wii") {
     formats.append("*.iso *.cso *.gcz *.wbfs *.rvz");
   } else if(platform == "wiiu") {
-    formats.append("*.iso *.cso *.gcz *.wbfs *.wud *.wux *.rpx *.app");
+    formats.append("*.iso *.cso *.gcz *.wbfs *.wua *.wud *.wux *.rpx *.app");
   } else if(platform == "wonderswan") {
     formats.append("*.ws");
   } else if(platform == "wonderswancolor") {
