--- /engine/Makefile
+++ /engine/Makefile
@@ -126,7 +126,7 @@
 LIBRARIES       = $(SDKPATH)/usr/lib
 CFLAGS          += -DAMD64
 # Workaround for GCC 9
-CFLAGS          +=  -Wno-error=stringop-overflow
+CFLAGS          +=  -Wno-error=stringop-overflow -Wno-error=address
 endif
 
 ifdef BUILD_LINUX_LE_arm
@@ -146,7 +146,7 @@
 OBJTYPE         = elf
 LIBRARIES       = $(SDKPATH)/usr/lib
 # Workaround for GCC 9
-CFLAGS          += -Wno-error=format-overflow -Wno-error=implicit-function-declaration -Wno-error=unused-variable -Wno-error=unused-label -Wno-error=stringop-overflow
+CFLAGS          += -Wno-error=format-overflow -Wno-error=implicit-function-declaration -Wno-error=unused-variable -Wno-error=unused-label -Wno-error=stringop-overflow -Wno-error=address
 endif
 
 ifdef BUILD_DARWIN

