From 4266f73c779971581fa7cc249ceb0f9c86353211 Mon Sep 17 00:00:00 2001
From: SupervisedThinking <supervisedthinking@gmail.com>
Date: Wed, 14 Sep 2022 15:17:59 +0200
Subject: [PATCH] Makefile: added / updated platform opts

---
 Makefile | 172 +++++++++++++++++++++++++++++++++----------------------
 1 file changed, 105 insertions(+), 67 deletions(-)

--- a/Makefile
+++ b/Makefile
@@ -1,14 +1,20 @@
 DEBUG = 0
-FORCE_GLES ?= 0
-FORCE_GLES3 ?= 0
-LLE ?= 0
+
+FORCE_GLES      ?= 0
+HAVE_GLES3      ?= 0
+HAVE_VULKAN     ?= 0
+HAVE_EGL_NO_X11 ?= 0
+
+FORCE_NO_PLUGINS  ?= 0
 HAVE_PARALLEL_RSP ?= 0
 HAVE_PARALLEL_RDP ?= 0
+HAVE_THR_AL       ?= 0
+LLE               ?= 0
 
 SYSTEM_MINIZIP ?= 0
-SYSTEM_LIBPNG ?= 0
-SYSTEM_XXHASH ?= 0
-SYSTEM_ZLIB ?= 0
+SYSTEM_LIBPNG  ?= 0
+SYSTEM_XXHASH  ?= 0
+SYSTEM_ZLIB    ?= 0
 
 HAVE_LTCG ?= 0
 DYNAFLAGS :=
@@ -96,24 +102,17 @@ ifneq (,$(findstring unix,$(platform)))
    ifeq ($(FORCE_GLES),1)
       GLES = 1
       GL_LIB := -lGLESv2
-   else ifeq ($(FORCE_GLES3),1)
+   else ifeq ($(HAVE_GLES3),1)
       GLES3 = 1
       GL_LIB := -lGLESv2
    else
       GL_LIB := -lGL
    endif
 
-   COREFLAGS += -DOS_LINUX
-   ifeq ($(ARCH), x86_64)
-      ASFLAGS = -f elf64 -d ELF_TYPE
-   else
-      ASFLAGS = -f elf -d ELF_TYPE
-   endif
-
+   # Generic ARM platforms
    ifneq (,$(findstring armv,$(platform)))
-      ARCH = arm
-      WITH_DYNAREC = arm
       CPUFLAGS += -DARM -marm
+      WITH_DYNAREC=arm     
       ifneq (,$(findstring cortexa8,$(platform)))
          CPUFLAGS += -mcpu=cortex-a8
       else ifneq (,$(findstring cortexa9,$(platform)))
@@ -121,33 +120,79 @@ ifneq (,$(findstring unix,$(platform)))
       else
          CPUFLAGS += -mcpu=cortex-a7
       endif
+      # NEON
       ifneq (,$(findstring neon,$(platform)))
-          CPUFLAGS += -mfpu=neon
-          HAVE_NEON = 1
+        CPUFLAGS += -mfpu=neon
+        HAVE_NEON = 1
       endif
+      # Target FPU
       ifneq (,$(findstring softfloat,$(platform)))
-          CPUFLAGS += -mfloat-abi=softfp
+         CPUFLAGS += -mfloat-abi=softfp
       else ifneq (,$(findstring hardfloat,$(platform)))
-          CPUFLAGS += -mfloat-abi=hard
+         CPUFLAGS += -mfloat-abi=hard
       endif
    endif
 
-# Raspberry Pi
-else ifneq (,$(findstring rpi,$(platform)))
-   TARGET := $(TARGET_NAME)_libretro.so
-   LDFLAGS += -shared -Wl,--version-script=$(LIBRETRO_DIR)/link.T -Wl,--no-undefined -ldl
-   ifeq ($(FORCE_GLES3),1)
-      GLES3 = 1
+   COREFLAGS += -DOS_LINUX
+   ifeq ($(ARCH), x86_64)
+      # RDP/RSP Plugins - cxd4 / paraLLEl RSP / Angrylion RDP Plus
+      ifneq ($(FORCE_NO_PLUGINS),1)
+         LLE = 1
+         HAVE_PARALLEL_RSP = 1
+         HAVE_THR_AL = 1
+         # RDP Plugin paraLLEl-RDP (Vulkan compute based)
+         ifeq ($(HAVE_VULKAN),1)
+            HAVE_PARALLEL_RDP = 1
+         endif
+      endif
+      ASFLAGS = -f elf64 -d ELF_TYPE
    else
-      GLES = 1
+      ASFLAGS = -f elf -d ELF_TYPE
    endif
+
+# Raspberry Pi 0(1)/2/3/4 - bcm2835/Mesa 3D - 32 & 64 bit userspace
+else ifneq (,$(findstring RPi,$(platform)))
+   TARGET := $(TARGET_NAME)_libretro.so
+   LDFLAGS += -shared -Wl,--version-script=$(LIBRETRO_DIR)/link.T -Wl,--no-undefined -ldl
+
+   # If Mesa 3D VC4/V3D graphics driver stack is used instead of Broadcoms bcm2835-driver
    ifneq (,$(findstring mesa,$(platform)))
       MESA = 1
    endif
-   ifneq (,$(findstring rpi4,$(platform)))
-      GLES3 = 1
-      MESA = 1
+
+   ifneq (,$(findstring RPi0,$(platform)))
+      CPUFLAGS += -mcpu=arm1176jzf-s -mfpu=vfp
+      GLES = 1
+      HAVE_NEON = 0
+   else
+      # RPi2/3/4 support NEON 
+      ifeq ($(ARCH), arm)
+         HAVE_NEON = 1
+      endif
+
+      ifneq (,$(findstring RPi2,$(platform)))
+         CPUFLAGS += -mcpu=cortex-a7 -mfpu=neon-vfpv4
+         GLES = 1
+      else
+         ifeq ($(ARCH), arm)
+            CPUFLAGS += -mfpu=neon-fp-armv8
+         endif
+
+         ifneq (,$(findstring RPi3,$(platform)))
+            CPUFLAGS += -mcpu=cortex-a53
+            GLES = 1
+         else ifneq (,$(findstring RPi4,$(platform)))
+            CPUFLAGS += -mcpu=cortex-a72
+            MESA = 1
+            ifeq ($(FORCE_GLES),1)
+               GLES = 1
+            else
+               GLES3 = 1
+            endif
+         endif
+      endif
    endif
+
    ifeq ($(MESA), 1)
       GL_LIB := -lGLESv2
    else
@@ -157,36 +202,15 @@ else ifneq (,$(findstring rpi,$(platform
       EGL_LIB := -lbrcmEGL
       INCFLAGS += -I/opt/vc/include -I/opt/vc/include/interface/vcos -I/opt/vc/include/interface/vcos/pthreads
    endif
-   HAVE_NEON = 1
-   ifneq (,$(findstring rpi2,$(platform)))
-      CPUFLAGS += -mcpu=cortex-a7
-      ARM_CPUFLAGS = -mfpu=neon-vfpv4
-   else ifneq (,$(findstring rpi3,$(platform)))
-      ifneq (,$(findstring rpi3_64,$(platform)))
-         CPUFLAGS += -mcpu=cortex-a53 -mtune=cortex-a53
-      else
-         CPUFLAGS += -march=armv8-a+crc -mtune=cortex-a53
-         ARM_CPUFLAGS = -mfpu=neon-fp-armv8
-      endif
-   else ifneq (,$(findstring rpi4,$(platform)))
-      ifneq (,$(findstring rpi4_64,$(platform)))
-         CPUFLAGS += -mcpu=cortex-a72 -mtune=cortex-a72
-      else
-         CPUFLAGS += -march=armv8-a+crc -mtune=cortex-a72
-         ARM_CPUFLAGS = -mfpu=neon-fp-armv8
-      endif
-   else ifneq (,$(findstring rpi,$(platform)))
-      CPUFLAGS += -mcpu=arm1176jzf-s
-      ARM_CPUFLAGS = -mfpu=vfp
-      HAVE_NEON = 0
-   endif
+
    ifeq ($(ARCH), aarch64)
       WITH_DYNAREC=aarch64
       HAVE_NEON = 0
    else
       WITH_DYNAREC=arm
-      CPUFLAGS += $(ARM_CPUFLAGS) -mfloat-abi=hard
+      CPUFLAGS += -mfloat-abi=hard
    endif
+
    COREFLAGS += -DOS_LINUX
    ASFLAGS = -f elf -d ELF_TYPE
 
@@ -297,20 +321,25 @@ else ifneq (,$(findstring AMLG,$(platfor
       else
          CPUFLAGS += -mtune=cortex-a53
       endif
-      GLES3 = 1
+
+      ifeq ($(FORCE_GLES),1)
+         GLES = 1
+      else
+         GLES3 = 1
+      endif
    else ifneq (,$(findstring AMLGX,$(platform)))
       CPUFLAGS += -mtune=cortex-a53
       ifneq (,$(findstring AMLGXM,$(platform)))
-         GLES3 = 1
+         ifeq ($(FORCE_GLES),1)
+            GLES = 1
+         else
+            GLES3 = 1
+         endif
       else
          GLES = 1
       endif
    endif
 
-   ifneq (,$(findstring mesa,$(platform)))
-      COREFLAGS += -DEGL_NO_X11
-   endif
-
    ifneq (,$(findstring mali,$(platform)))
       GL_LIB := -lGLESv3
    else
@@ -365,18 +394,22 @@ else ifneq (,$(findstring RK,$(platform)
       CPUFLAGS += -march=armv8-a+crc -mfloat-abi=hard -mfpu=neon-fp-armv8
       ifneq (,$(findstring RK3399,$(platform)))
          CPUFLAGS += -mtune=cortex-a72.cortex-a53
-         GLES3 = 1
+         ifeq ($(FORCE_GLES),1)
+            GLES = 1
+         else
+            GLES3 = 1
+         endif
       else ifneq (,$(findstring RK3328,$(platform)))
          CPUFLAGS += -mtune=cortex-a53
          GLES = 1
       endif
    else ifneq (,$(findstring RK3288,$(platform)))
       CPUFLAGS += -march=armv7ve -mtune=cortex-a17 -mfloat-abi=hard -mfpu=neon-vfpv4
-      GLES3 = 1
-   endif
-
-   ifneq (,$(findstring mesa,$(platform)))
-      COREFLAGS += -DEGL_NO_X11
+      ifeq ($(FORCE_GLES),1)
+         GLES = 1
+      else
+         GLES3 = 1
+      endif
    endif
 
    GL_LIB := -lGLESv2
@@ -417,7 +450,7 @@ else ifneq (,$(findstring ios,$(platform
 		WITH_DYNAREC=
 		GLES=1
 		GLES3=1
-		FORCE_GLES3=1
+		HAVE_GLES3=1
 		EGL := 0
 		PLATCFLAGS += -DHAVE_POSIX_MEMALIGN -DNO_ASM
 		PLATCFLAGS += -DIOS -marm -DOS_IOS -DDONT_WANT_ARM_OPTIMIZATIONS
@@ -558,6 +591,11 @@ ifeq ($(HAVE_NEON), 1)
    COREFLAGS += -DHAVE_NEON -D__ARM_NEON__ -D__NEON_OPT -ftree-vectorize -mvectorize-with-neon-quad -ftree-vectorizer-verbose=2 -funsafe-math-optimizations -fno-finite-math-only
 endif
 
+# Define EGL platform
+ifeq ($(HAVE_EGL_NO_X11), 1)
+  COREFLAGS += -DEGL_NO_X11
+endif
+
 ifeq ($(LLE), 1)
    COREFLAGS += -DHAVE_LLE
 endif
