--- a/bsnes/GNUmakefile
+++ b/bsnes/GNUmakefile
@@ -1,7 +1,7 @@
 target ?= bsnes
 binary ?= application
 build := performance
-openmp := true
+openmp := false
 local ?= true
 flags += -I. -I..
 
--- a/bsnes/target-libretro/GNUmakefile
+++ b/bsnes/target-libretro/GNUmakefile
@@ -1,6 +1,6 @@
 name := bsnes_libretro
 local := false
-openmp := true
+openmp := false
 flags += -Wno-narrowing -Wno-multichar -g -fPIC
 
 ifeq ($(platform), ios-arm64)
@@ -24,11 +24,11 @@
 
 all: $(objects)
 ifneq ($(filter $(platform),linux bsd unix unix-armv7-hardfloat-neon),)
-	$(strip $(compiler) -o out/bsnes_libretro.so -shared $(objects) -Wl,--no-undefined -Wl,--version-script=target-libretro/link.T -lgomp -Wl,-Bdynamic $(options))
+	$(strip $(compiler) -o out/bsnes_libretro.so -shared $(objects) -Wl,--no-undefined -Wl,--version-script=target-libretro/link.T -Wl,-Bdynamic $(options))
 else ifeq ($(platform),haiku)
-	$(strip $(compiler) -o out/bsnes_libretro.so -shared $(objects) -Wl,--no-undefined -Wl,--version-script=target-libretro/link.T -lgomp -Wl,-Bdynamic $(options))
+	$(strip $(compiler) -o out/bsnes_libretro.so -shared $(objects) -Wl,--no-undefined -Wl,--version-script=target-libretro/link.T -Wl,-Bdynamic $(options))
 else ifeq ($(platform),windows)
-	$(strip $(compiler) -o out/bsnes_libretro.dll -shared $(objects) -Wl,--no-undefined -Wl,--version-script=target-libretro/link.T -lgomp -Wl,-Bdynamic $(options))
+	$(strip $(compiler) -o out/bsnes_libretro.dll -shared $(objects) -Wl,--no-undefined -Wl,--version-script=target-libretro/link.T -Wl,-Bdynamic $(options))
 else ifeq ($(platform),libnx)
 	$(strip $(AR) rcs out/bsnes_libretro_libnx.a $(objects))
 else ifeq ($(platform),macos)

